components/script/dom/gputexture.rs

```diff
@@ -152,31 +153,38 @@ impl GPUTextureMethods for GPUTexture {
                 GPUTextureAspect::Depth_only => wgt::TextureAspect::DepthOnly,
             },
             base_mip_level: descriptor.baseMipLevel,
-            level_count: if descriptor.mipLevelCount == 0 {
-                self.mip_level_count - descriptor.baseMipLevel
-            } else {
-                descriptor.mipLevelCount
-            },
+            level_count: descriptor.mipLevelCount.map_or(
```

Comment:
Actually, you don't need to compute this. Passing `0` down to `wgpu-core` would mean the same.
Perhaps, this and the layer count computation should look like:
```rust
level_count: match descriptor.mipLevelCount {
  Some(count) if count != 0 && descriptor.baseMipLevel + count <= self.mip_level_count => NonZeroU32::new(count),
  None if descriptor.baseMipLevel < self.mip_level_count => None,
  _ => { generate an error },
},
```

Reply:
I didn't do this because we won't be sending a `createTextureView` message to server in last case and an error Id won't be registered.

Reply:
I think `descriptor.mapLevelCount.and_then(NonZeroU32::new)` would work, since the validation is done in wgpu-core

Reply:
ok, so here we want to send *something* to wgpu-core, right?
`descriptor.mipLevelCount.and_then(|count| NonZero::new(if count == 0 { !0 } else {count})`?

The problem with the current code is that `0` count will turn into None and result in automatic mipmapping, for example, while the expected outcome is an error